grabbit/src/main/groovy/com/twcable/grabbit/client/servlets/GrabbitJobServlet.groovy

```diff
@@ -0,0 +1,170 @@
+                    response.setStatus(SC_BAD_REQUEST)
+                    response.writer.write("Request was made to get the status of a job, but a job could not be found for the ID provided for ${request.pathInfo}")
+                    return
+                }
+            }
+        }
+        log.warn "Bad request to receive job status for : ${request.pathInfo}. A job ID was not provided in the request."
+        response.setStatus(SC_BAD_REQUEST)
+        response.writer.write("Request was made to get the status of a job, but no job ID was provided for ${request.pathInfo}")
+    }
+
+    /**
+     * Used to create a new set of jobs from a Grabbit configuration file
+     *
+     * @param request the request to create a new set of jobs
+     * @param response Will respond with a HTTP 200 if the configuration was accepted, and jobs started; A JSON array of jobIDs will be returned in the body, and a transactionId
+     * in the response header, with the key "GRABBIT_TRANSACTION_ID." If no configuration is sent with the request, a bad request response will be returned. A bad request response
+     * will also be returned if the configuration provided is rejected. See {@link com.twcable.grabbit.GrabbitConfiguration} to see why this might happen.
+     */
+    @Override
+    protected void doPut(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+        if(!request.inputStream) {
+            response.setStatus(SC_BAD_REQUEST)
+            response.writer.write("Sorry, we couldn't create your jobs. It looks like a Grabbit configuration file is missing from the request.")
+            return
+        }
+        final configurationInput = IOUtils.toString(request.inputStream, request.characterEncoding)
+        log.debug "Input: ${configurationInput}"
+
+        //The Login of the user making this request.
+        //This user will be used to connect to JCR
+        //If the User is null, 'anonymous' will be used to connect to JCR
+        final clientUsername = request.remoteUser
+        final GrabbitConfiguration configuration
+        try {
+            configuration = GrabbitConfiguration.create(configurationInput)
+        } catch(ConfigurationException ex) {
+            log.warn "Bad configuration for request. ${ex.errors.values().join(',')}"
+            response.status = SC_BAD_REQUEST
+            response.writer.write("Bad configuration for request. More detail: \n\n ${new JsonBuilder(ex.errors).toString()}")
+            return
+        }
+        Collection<Long> jobExecutionIds = clientService.initiateGrab(configuration, clientUsername)
+        log.info "Jobs started : ${jobExecutionIds}"
+        log.info "Transaction started ${configuration.transactionID}"
+        response.status = SC_OK
+        response.contentType = "application/json"
+        response.addHeader("GRABBIT_TRANSACTION_ID", "${configuration.transactionID}")
```

Comment:
Kind of unrelated...looking at this line, may want to change some of these lines to avoid the GString wrap/unwrap pattern